src/fsharp/TastOps.fs

```diff
@@ -1776,19 +1776,17 @@ let isRefTy g ty =
         isUnitTy g ty
     )
 
-// ECMA C# LANGUAGE SPECIFICATION, 27.2
-// An unmanaged-type is any type that isn't a reference-type, a type-parameter, or a generic struct-type and
+// An unmanaged-type is any type that isn't a reference-type and
 // contains no fields whose type is not an unmanaged-type. In other words, an unmanaged-type is one of the
 // following:
 // - sbyte, byte, short, ushort, int, uint, long, ulong, char, float, double, decimal, or bool.
 // - Any enum-type.
 // - Any pointer-type.
-// - Any non-generic user-defined struct-type that contains fields of unmanaged-types only.
-// [Note: Constructed types and type-parameters are never unmanaged-types. end note]
+// - Any generic user-defined struct-type that can be statically determined to be 'unmanaged' at construction.
 let rec isUnmanagedTy g ty =
     let ty = stripTyEqnsAndMeasureEqns g ty
-    match tryDestAppTy g ty with
-    | ValueSome tcref ->
+    match ty with
+    | TType_app(tcref, tinst) ->
```

Comment:
You should almost never match on `TType` directly - always use `tryAppTy` (note, `tryDestAppTy` should really be called `tryTcrefOfAppTy`)

Reply:
The reason you should never match on `TType` is that it might be a solved type variable or a type abbreviation, both of which get expanded by `tryAppTy` and friends

Reply:
That's fair. I thought it was fine considering we were stripping eqns anyway.

Reply:
Yes, you're correct